### ISSUE TITLE
Golangci-lint v2 erroring due to hardcoded flag thats no longer supported in vscode

### ISSUE THREAD

**Issue by jacobdrury** [golangci/golangci-lint#5640]
### Welcome

- [x] Yes, I'm using a binary release within 2 latest [releases](https://github.com/golangci/golangci-lint/releases). Only such installations are supported.
- [x] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [x] Yes, I've read the `typecheck` section of the [FAQ](https://golangci-lint.run/welcome/faq/#why-do-you-have-typecheck-errors).
- [x] Yes, I've tried with the standalone [linter](https://golangci-lint.run/usage/linters/) if available (e.g., gocritic, go vet, etc.).
- [x] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### How did you install golangci-lint?

go install

### Description of the problem

After migrating to v2.0.2, I am getting an error in the vscode output console when running the linter. It seems that vscode is always including `--print-issued-lines=false` in the command which is no longer supported. This breaks the vscode integration. Is there a setting I can change to tell vscode to not include this flag or some other work around?

settings.json
```json
    "go.lintTool": "golangci-lint",
    "go.lintFlags": [
        "--fast-only"
    ],
    "go.formatTool": "custom",
    "go.alternateTools": {
    "customFormatter": "golangci-lint"
    },
    "go.formatFlags": [
        "fmt",
        "--stdin"
    ]
```

output:
```
2025-03-26 13:42:18.833 [error] Error while running tool: /Users/jacobdrury/go/bin/golangci-lint run --fast-only --print-issued-lines=false --out-format=colored-line-number --issues-exit-code=0
2025-03-26 13:42:18.834 [error] Error: unknown flag: --print-issued-lines
Failed executing command with error: unknown flag: --print-issued-lines
```

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version v2.0.2 built with go1.23.2 from (unknown, modified: ?, mod sum: "h1:dMCC8ikPiLDvHMFy3+XypSAuGDBOLzwWqqamer+bWsY=") on (unknown)
```

</details>

### Configuration

<details>

```console
version: "2"
run:
  allow-parallel-runners: true
output:
  formats:
    text:
      path: stdout
    junit-xml:
      path: lint-report.xml
linters:
  enable:
    - asciicheck
    - copyloopvar
    - errname
    - exptostd
    - forcetypeassert
    - ginkgolinter
    - gocheckcompilerdirectives
    - godox
    - gosec
    - intrange
    - makezero
    - misspell
    - nlreturn
    - nonamedreturns
    - prealloc
    - unconvert
    - unparam
    - usestdlibvars
    - wastedassign
    - whitespace
    - wsl
  settings:
    ginkgolinter:
      suppress-len-assertion: false
      suppress-nil-assertion: false
      suppress-err-assertion: false
      suppress-compare-assertion: false
      suppress-async-assertion: false
      suppress-type-compare-assertion: false
      forbid-focus-container: true
      allow-havelen-zero: false
      force-expect-to: true
      validate-async-intervals: true
      forbid-spec-pollution: true
      force-succeed: true
    godox:
      keywords:
        - TODO
    misspell:
      mode: restricted
      locale: US
      ignore-rules:
        - cancelled
        - aircrafts
    nlreturn:
      block-size: 1
    staticcheck:
      checks:
        - S1000
        - S1001
        - S1002
        - S1003
        - S1004
        - S1005
        - S1006
        - S1007
        - S1008
        - S1009
        - S1010
        - S1011
        - S1012
        - S1017
        - S1018
        - S1019
        - S1020
        - S1021
        - S1023
        - S1024
        - S1025
        - S1028
        - S1029
        - S1030
        - S1031
        - S1032
        - S1033
        - S1034
        - S1035
        - S1036
        - S1037
        - S1038
        - S1039
        - S1040
    whitespace:
      multi-if: false
      multi-func: false
    wsl:
      strict-append: true
      allow-assign-and-call: true
      allow-assign-and-anything: true
      allow-multiline-assign: true
      force-case-trailing-whitespace: 0
      allow-trailing-comment: true
      allow-separated-leading-comment: true
      allow-cuddle-declarations: false
      allow-cuddle-with-calls:
        - Lock
        - RLock
      allow-cuddle-with-rhs:
        - Unlock
        - RUnlock
      force-err-cuddling: true
      error-variable-names:
        - err
      force-short-decl-cuddling: false
      # Controls if you may cuddle variables used anywhere in the following block.
      # https://github.com/bombsimon/wsl/blob/master/doc/configuration.md#allow-cuddle-used-in-block
      # Default: false
      allow-cuddle-used-in-block: true
  exclusions:
    presets:
      - comments
      - common-false-positives
      - legacy
      - std-error-handling
    paths:
      - third_party$
      - builtin$
      - examples$
issues:
  new-from-merge-base: main
  whole-files: true
formatters:
  enable:
    - gofmt
  settings:
    gofmt:
      simplify: true
      rewrite-rules:
        - pattern: interface{}
          replacement: any
        - pattern: a[b:len(a)]
          replacement: a[b:]
        - pattern: (a)
          replacement: a
  exclusions:
    paths:
      - third_party$
      - builtin$
      - examples$

```

</details>

### Go environment

<details>

```console
$ go version && go env
# paste output here
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
# paste output here
```

</details>

### A minimal reproducible example or link to a public repository

<details>

```go
// add your code here
```

</details>

### Validation

- [x] Yes, I've included all information above (version, config, etc.).

### Supporter

- [ ] I am a sponsor/backer through [GitHub](https://github.com/sponsors/golangci/) or [OpenCollective](https://opencollective.com/golangci-lint)

**Comment by alexandear** [comment]
Hi @jacobdrury,

The answer is here: https://github.com/golangci/golangci-lint/issues/5616#issuecomment-2750891266

**Comment by adamk33n3r** [comment]
I installed the pre-release version of the extension but I'm still getting this error

**Comment by ldez** [comment]
If you have this error, it is because you are using golangci-lint v1.

https://golangci-lint.run/welcome/integrations/#visual-studio-code

**Comment by adamk33n3r** [comment]
I'm not sure how that could be true
```
2025-05-26 19:48:33.371 [info] Running checks...
2025-05-26 19:48:33.426 [info] Starting linting the current package at d:\dev\NoMoreBorderGo
2025-05-26 19:48:33.605 [error] Error while running tool: C:\ProgramData\chocolatey\bin\golangci-lint.exe run --path-mode=abs --fast-only --issues-exit-code=0 --print-issued-lines=false --out-format=colored-line-number
2025-05-26 19:48:33.605 [error] Error: unknown flag: --print-issued-lines
Failed executing command with error: unknown flag: --print-issued-lines
```

```
PS > C:\ProgramData\chocolatey\bin\golangci-lint.exe --version
golangci-lint has version 2.1.6 built with go1.24.2 from eabc2638 on 2025-05-04T15:41:19Z
```

**Comment by ldez** [comment]
Your vscode is using your local version, but vscode thinks you are using v1.

The extension uses the following code to detect the version:
```
go version -m C:\ProgramData\chocolatey\bin\golangci-lint.exe
```

It takes the 2nd line, splits on `\s`, and uses the 3rd part (the version).

**Comment by ldez** [comment]
If you still have the problem, can you open a dedicated discussion?

And provide the following information:

```bash
code --version 
```

```bash
code --list-extensions --show-versions | grep 'golang.go@'
# OR if you don't have grep
code --list-extensions --show-versions
```

```bash
golangci-lint.exe --version
```

```bash
go version
```

https://github.com/golangci/golangci-lint/discussions/categories/q-a
